src/components/RecentSnippets.jsx

```diff
@@ -8,24 +8,73 @@ import * as actions from '../actions';
 import '../styles/RecentSnippets.styl';
 
 class RecentSnippets extends React.Component {
+  constructor(props) {
+    super(props);
+    this.nextSetOfSnippets = this.nextSetOfSnippets.bind(this);
+    this.prevSetOfSnippets = this.prevSetOfSnippets.bind(this);
+  }
+
   componentDidMount() {
-    const { dispatch } = this.props;
-    dispatch(actions.fetchRecentSnippets);
+    const { dispatch, recent, pagination } = this.props;
+    let marker = null;
+
+    if (pagination.get('prev')) {
+      marker = recent.get(0) + 1;
```

Comment:
Hmm, I'm not sure what this is doing. We can get the marker value from the URLs in `Link`. Why do we need to compute it here?

Reply:
from `Link` we get next marker and previous marker, but we don't have current marker in case I want to click 'Show' on tile and on return I want to be on the same page I was before.If someone sees a better way to handle this, I would be glide to change